examples/HILO_Symbol_Plotting.py

```diff
@@ -0,0 +1,158 @@
+"""
+
+###############################
+# Imports
+import numpy as np
+import matplotlib.pyplot as plt
+from metpy.units import units
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+from datetime import datetime
+from netCDF4 import num2date
+from siphon.ncss import NCSS
+from scipy.ndimage import gaussian_filter
+
+###############################
+# Function for finding and plotting max/min points
+
+
+def plot_HILO_points(lon, lat, data, hilo, nsize, transform=None):
+    '''
+    This function will find and plot relative maximum and minimum for a 2D grid. The function
+    is designed to plot an 'H' for maximum values (e.g., High pressure) and an 'L' for minimum
+    values (e.g., low pressue). It first uses a one standard deviation gaussian filter to find
+    a synoptic scale HILO value, then uses the scipy.ndimage filters for maximum or minimum
+    values, finally the HILO symbol is plotted on the current axes with the smoothed integer
+    value of the HILO.
+
+    lon = plotting longitude values (2D)
+    lat = plotting latitude values (2D)
+    data = 2D data that you wish to plot the HILO placement
+    hilo = Either a value of 'hi' for Maximum Values or 'lo' for Minimum Values
+    nsize = Size of the grid box to filter the max and min values to plot a reasonable number
+
+    The H and L symbol will be plotted on the current axes within the bounding frame
+    (e.g., clip_on=True)
+    '''
+    from scipy.ndimage.filters import maximum_filter, minimum_filter
+
+    if (hilo == 'hi'):
+        data_ext = maximum_filter(data, nsize, mode='nearest')
+        txt_ext = 'H'
+        color = 'b'
+    elif (hilo == 'lo'):
+        data_ext = minimum_filter(data, nsize, mode='nearest')
+        txt_ext = 'L'
+        color = 'r'
+
+    mxy, mxx = np.where(data_ext == data)
+
+    for i in range(len(mxy)):
+        ax.text(lon[mxy[i], mxx[i]], lat[mxy[i], mxx[i]], txt_ext, color=color, size=24,
+                clip_on=True, horizontalalignment='center', verticalalignment='center',
+                transform=transform)
+        ax.text(lon[mxy[i], mxx[i]], lat[mxy[i], mxx[i]], '\n'+str(np.int(data[mxy[i], mxx[i]])),
+                color=color, size=12, clip_on=True, fontweight='bold',
+                horizontalalignment='center', verticalalignment='top', transform=transform)
+
+
+###############################
+# Get NARR data
+dattim = datetime(1999, 1, 3, 0)
+
+ncss = NCSS('http://nomads.ncdc.noaa.gov/thredds/ncss/grid/narr-a/{0:%Y%m}/{0:%Y%m%d}/'
+            'narr-a_221_{0:%Y%m%d}_{0:%H}00_000.grb'.format(dattim))
+query = ncss.query()
+query.all_times().variables('Pressure_reduced_to_MSL',
+                            'Geopotential_height').add_lonlat().accept('netcdf4')
+data = ncss.get_data(query)
+
+###############################
+# Extract data into variables
+
+# Grab pressure levels
+plev = list(data.variables['isobaric'][:])
+
+# Grab lat/lons and make all lons 0-360
```

Comment:
Maybe we should have a function in calc to do this? Thoughts?

Reply:
That might be a really good thing. I really like making this a list to use the `.index()` method to find a particular pressure level. Looks nice and clean.